### DISCUSSION TITLE
VirusTotal URL scan

### DISCUSSION THREAD

**Discussion by Pato98** [rspamd/rspamd#4722] [General]
Can I use the VirusTotal API for URL scan with Rspamd?
We already use Eset Server Security with ICAP and it's working fine with attachments, but we receive a lot of phishing URL and the VirusTotal looks great solution for that problem.

**Comment by fatalbanana** [comment]
It seems to be readily supported by [external services module](https://rspamd.com/doc/modules/external_services.html#virustotal-details). Ratelimits are a likely issue so it's important that one should have an unencumbered API key to use it.

**Reply by Bierchermuesli** [reply]
no, [this](https://github.com/rspamd/rspamd/blob/master/lualib/lua_scanners/virustotal.lua) is for file checks and not for url checks
